src/main/scala/ldbc/snb/datagen/spark/generators/SparkPersonSerializer.scala

```diff
@@ -0,0 +1,126 @@
+      val dynamicPersonSerializer = conf.getDynamicPersonSerializer
+      val hadoopConf = serializableHadoopConf.value
+      val partitionId = TaskContext.getPartitionId()
+      val buildDir = conf.getBuildDir
+
+      val fs = FileSystem.get(hadoopConf)
+      fs.mkdirs(new Path(buildDir))
+
+      dynamicPersonSerializer.initialize(
+        hadoopConf,
+        conf.getSocialNetworkDir,
+        partitionId,
+        conf.isCompressed,
+        conf.insertTrailingSeparator()
+      )
+
+      var insertEventSerializer: InsertEventSerializer = null
+      var deleteEventSerializer: DeleteEventSerializer = null
+
+      if ((DatagenParams.getDatagenMode eq DatagenMode.INTERACTIVE) || (DatagenParams.getDatagenMode eq DatagenMode.BI)) {
+        insertEventSerializer = new InsertEventSerializer(hadoopConf, conf.getBuildDir + "/temp_insertStream_person_" + partitionId, partitionId, DatagenParams.numUpdateStreams)
+        deleteEventSerializer = new DeleteEventSerializer(hadoopConf, conf.getBuildDir + "/temp_deleteStream_person_" + partitionId, partitionId, DatagenParams.numUpdateStreams)
+      }
+
+      try {
+        DatagenContext.initialize(conf)
+
+        for {p <- persons} {
+          if (DatagenParams.getDatagenMode eq DatagenMode.RAW_DATA) {
+            dynamicPersonSerializer.export(p)
+            for (k <- p.getKnows.iterator.asScala) {
+              dynamicPersonSerializer.export(p, k)
+            }
+          }
+          else {
+            if (p.getCreationDate < Dictionaries.dates.getBulkLoadThreshold && (p.getDeletionDate >= Dictionaries.dates.getBulkLoadThreshold && p.getDeletionDate <= Dictionaries.dates.getSimulationEnd)) {
+              dynamicPersonSerializer.export(p)
+              if (p.isExplicitlyDeleted) {
+                deleteEventSerializer.export(p)
+                deleteEventSerializer.changePartition()
+              }
+            }
+            else if (p.getCreationDate < Dictionaries.dates.getBulkLoadThreshold && p.getDeletionDate > Dictionaries.dates.getSimulationEnd) dynamicPersonSerializer.export(p)
+            else if (p.getCreationDate >= Dictionaries.dates.getBulkLoadThreshold && (p.getDeletionDate >= Dictionaries.dates.getBulkLoadThreshold) && p.getDeletionDate <= Dictionaries.dates.getSimulationEnd) {
+              insertEventSerializer.export(p)
+              insertEventSerializer.changePartition()
+              if (p.isExplicitlyDeleted) {
+                deleteEventSerializer.export(p)
+                deleteEventSerializer.changePartition()
+              }
+            }
+            else if (p.getCreationDate >= Dictionaries.dates.getBulkLoadThreshold && p.getDeletionDate > Dictionaries.dates.getSimulationEnd) {
+              insertEventSerializer.export(p)
+              insertEventSerializer.changePartition()
+            }
+            //TODO: export was split between here and HadoopPersonActivityGenerator, not sure why
+            // moved all here
```

Comment:
IMHO it was quite arbitrary. Previously, you could mix e.g. a `CsvBasic` serializer for persons and a `CsvMergeForeign` serializer for activities. No one used it that way though.

Jack changed a couple of things around here regarding insert/delete events but the version in this PR should be OK.

Reply:
note: I copy-pasted this code and auto-converted it to Scala.
This part could use a heavy refactor...